### PR TITLE
fix: [CDS-54502]: Make TriggerExecutionHelper use ExecutionHelper methods for starting execution 

### PR DESCRIPTION
**Pull request by vcalasansh**
## Describe your changes
This PR proposes a refactor to `createPlanExecution` method in `TriggerExecutionHelper` class: we are trying to use methods from `ExecutionHelper` as much as possible for fetching execution arguments and populating data necessary for triggering a pipeline execution, which are the same methods used when execution is started manually either via UI or API.

This will make trigger execution much more consistent with executions triggered manually (same error handling and validation logic for inputs), as well as reduce risk of issues showing up every time an update to pipeline execution flow  code takes place and is not accordingly updated in Triggers code. Also, most of execution logic will now be centered only in `ExecutionHelper` instead of having a lot of repeated code in both `TriggerExecutionHelper` and `ExecutionHelper`.

This refactor is initially behind a FF called `CDS_NG_TRIGGER_EXECUTION_REFACTOR`, which we plan to slowly rollout first in our test/internal accounts, then for customers in case no issues appear. 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45385)

### CODE REVIEW

File: pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java [hunk 1]
```diff
@@ -559,6 +678,17 @@ public String fetchInputSetYAML(TriggerDetails triggerDetails, TriggerWebhookEve
     return mergeInputSetResponseDTOPMS.getPipelineYaml();
   }
 
+  public String fetchInputSetYAMLV2(
+      PipelineEntity pipelineEntity, TriggerDetails triggerDetails, TriggerWebhookEvent triggerWebhookEvent) {
```
Comment (rathodmeetsatish) [inline pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java:682]: we decided to remove this part from create plan execution and methods calling create plan execution should pass runtimeinputyaml

**Review by sribalij — DISMISSED** [review]
> Reviewed 4 of 5 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45385)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by rathodmeetsatish — APPROVED** [review]
> Reviewed 1 of 5 files at r1, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45385)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Review by sribalij — APPROVED** [review]
> Reviewed 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45385)* status: :shipit: complete! ✓

**Comment by vcalasansh** [comment]
trigger messagecheck

**Comment by vcalasansh** [comment]
trigger messagecheck

**Pull request merged**